app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -24,37 +24,55 @@ class PromotedStoryListView @JvmOverloads constructor(
 
   @Inject
   lateinit var bindingInterface: ViewBindingShim
+  private var dataList: List<PromotedStoryViewModel> = listOf()
 
   override fun onAttachedToWindow() {
     super.onAttachedToWindow()
 
     (FragmentManager.findFragment<Fragment>(this) as ViewComponentFactory)
       .createViewComponent(this).inject(this)
 
-    adapter = BindableAdapter.SingleTypeBuilder.newBuilder<PromotedStoryViewModel>()
+    /**
+     * The StartSnapHelper is used to snap between items rather than smooth scrolling, so that
+     * the item is completely visible in [HomeFragment] as soon as learner lifts the finger
+     * after scrolling.
+     */
+    val snapHelper = StartSnapHelper()
+    this.onFlingListener = null
+    snapHelper.attachToRecyclerView(this)
+  }
+
+  /**
+   * Sets the list of promoted stories that this view shows to the learner.
+   *
+   * @param newDataList the new list of stories to present
+   */
+  fun setDataList(newDataList: List<PromotedStoryViewModel>) {
```

Comment:
Should this list be marked as nullable if we anticipate it can be null?

Reply:
Actually I don't believe this can ever be null:
1. The list is initialized to empty in the protobuf and seems unlikely to be set to null if an empty list represents no stories
2. A null / empty list shouldn't create a view model for this view so this should never happen.

Will add throw an error here instead

Reply:
Ah, I think databinding can let it be null under some circumstances (such as if the binding code in the layout file resolves to null), so we have to explicitly handle that case like we do in the shared binding adapter for BindableAdapter.

Reply:
As noted above we check for null & empty and give a warning.